src/org/zaproxy/zap/extension/pscanrulesAlpha/CacheableScanner.java

```diff
@@ -498,7 +499,7 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
 				lifetimeFound = true;
 				lifetime = 60 * 60 * 24 * 365;
 				//a liberal heuristic was assumed, for which no actual evidence exists
-				freshEvidence = Constant.messages.getString(MESSAGE_PREFIX_STORABLE_CACHEABLE + "evidence.liberallifetimeheuristic");
```

Comment:
(Not sure which line are you referring to, the link does not go to a specific line.)
There's at least one case that would lead to wrong evidence (when expiresHeadersFound is greater than 1, note that the freshEvidence is not reset in that branch). In any case it's less error prone if it's explicitly set to null (or an empty String).